### PR TITLE
#861 Updates for latest numpy and astropy versions

### PR DESCRIPTION
**Pull request by rmjarvis**
Numpy version 1.12 changed how `np.roll` works slightly, which broke one of our tests.  To make sure `galsim.utilities.roll2d` always returns a C-contiguous array, we now have to coerce it.

And astropy version 1.3 changes the name of the `clobber` kwarg for `writeto` to `overwrite`.  I have no idea what compelled them to make this change, but I can't be bothered to support both APIs, so I just implement the file removal ourselves in the two places that had used `clobber`.

### CODE REVIEW

File: tests/galsim_test_helpers.py [hunk 1]
```diff
@@ -435,14 +435,16 @@ def do_pickle(obj1, func = lambda x : x, irreprable=False):
             # Special case: can't change size of LVector or PhotonArray without changing array
             if classname in ['LVector', 'PhotonArray'] and i == 0:
                 continue
-            with galsim.utilities.printoptions(precision=18, threshold=1e6):
+            with galsim.utilities.printoptions(precision=18, threshold=np.inf):
                 try:
                     if classname in galsim._galsim.__dict__:
-                        obj6 = eval('galsim._galsim.' + classname + repr(tuple(newargs)))
+                        eval_str = 'galsim._galsim.' + classname + repr(tuple(newargs))
```
Comment (jmeyers314) [inline tests/galsim_test_helpers.py]: Why this change?  Were we catching the wrong exception?  If so, then maybe move the `try:` down below the `if-else`.

Comment (rmjarvis) [inline tests/galsim_test_helpers.py]: No.  The previous code was fine when everything worked.  I just wanted to be able to print the `eval_str` below when this failed in order to help figure out what the problem was with the `eval`.
